bundles/InstallBundle/src/Event/BundleSetupEvent.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Pimcore\Bundle\InstallBundle\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+class BundleSetupEvent extends Event
+{
+    private array $bundles;
+    private array $recommendations;
+
+    public function __construct(array $bundles, array $recommendations)
+    {
+       $this->bundles = $bundles;
+       $this->recommendations = $recommendations;
+    }
+
+    public function getBundles(): array
+    {
+        return $this->bundles;
+    }
+
+    public function getRecommendedBundles(): array
+    {
+        return $this->recommendations;
+    }
+
+    public function addBundle(string $key, string $class) : void
+    {
+        $this->bundles[$key] = $class;
+    }
+
+    public function removeBundle(string $key): void
+    {
+        if(array_key_exists($key, $this->bundles)) {
+            unset($this->bundles[$key]);
+        }
```

Comment:
I don't think this check is necessary. You can unset non-existent keys without an error/notice.

```suggestion
        unset($this->bundles[$key]);
```